src/trio/_core/_instrumentation.py

```diff
@@ -2,23 +2,25 @@
 
 import logging
 import types
-from collections.abc import Callable, Sequence
-from typing import TypeVar
+from typing import TYPE_CHECKING, TypeVar
 
 from .._abc import Instrument
 
 # Used to log exceptions in instruments
 INSTRUMENT_LOGGER = logging.getLogger("trio.abc.Instrument")
 
+if TYPE_CHECKING:
+    from collections.abc import Callable, Sequence
 
-# Explicit "Any" is not allowed
-F = TypeVar("F", bound=Callable[..., object])  # type: ignore[misc]
+    from typing_extensions import ParamSpec
+
+    P = ParamSpec("P")
+    T = TypeVar("T")
 
 
 # Decorator to mark methods public. This does nothing by itself, but
 # trio/_tools/gen_exports.py looks for it.
-# Explicit "Any" is not allowed
-def _public(fn: F) -> F:  # type: ignore[misc]
+def _public(fn: Callable[P, T]) -> Callable[P, T]:
```

Comment:
This should not be `ParamSpec`. There's a subtle difference between this and a callable-bound `TypeVar` - `ParamSpec` can't capture an overload since the return type might change, but the typevar form preserves it, so it would be passed through unchanged.

Reply:
I looked at all the uses of `@_public`. There are no overloaded functions.

Reply:
I think the focus is that there could be in the future

Reply:
Sure. Seems easy enough to deal with it then :-)

Reply:
Maybe, but there's no real cost to doing it correctly now, and prevents a confusing issue coming up later?